### PR TITLE
Add CloseButtonTooltip to the 'X' button on a SnackBar

### PR DESCRIPTION
**Pull request by dsanagustin**
Adds a localized Close Button tooltip to the 'X' Button on a SnackBar, making it readable by screen readers.

Github Issue #143793 

*If you had to change anything in the [flutter/tests] repo, include a link to the migration guide as per the [breaking change policy].*

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [ ] I signed the [CLA]. **(No need. I am a Googler.)**
- [x] I listed at least one issue that this PR fixes in the description above.
- [ ] I updated/added relevant documentation (doc comments with `///`).  **No need. I made a one-line change to a function.**
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [ ] I followed the [breaking change policy] and added [Data Driven Fixes] where supported.
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat
[Data Driven Fixes]: https://github.com/flutter/flutter/wiki/Data-driven-Fixes

### CODE REVIEW

File: packages/flutter/test/material/snack_bar_test.dart
Comment (goderbauer) [inline packages/flutter/test/material/snack_bar_test.dart:3271]: nit: Presumably, it animates in here?

```suggestion
    await tester.pumpAndSettle(); // Have the SnackBar fully animate in.
```

File: packages/flutter/test/material/snack_bar_test.dart
Comment (goderbauer) [inline packages/flutter/test/material/snack_bar_test.dart:3247]: Per convention, could you add a comment here linking to the issue?

```suggestion
    // Regression test for https://github.com/flutter/flutter/issues/143793
    await tester.pumpWidget(MaterialApp(
```

**Review by goderbauer — APPROVED** [review]
LGTM with two tiny nits

**Comment by dsanagustin** [comment]
> LGTM with two tiny nits

@goderbauer Suggestions applied, thank you so much! Quick question: I don't have write access to this repo -- how can I merge this PR once the checks pass?

**Review by gspencergoog — APPROVED** [review]
32384589-a60f0e74-c078-11e7-9bc1-e5b5287aea9d

**Pull request merged**
